Indico/Jobs/Job.cs

```diff
@@ -118,38 +119,28 @@ public async Task<JobStatus> Status()
         /// Retrieve result. Status must be success or an error will be thrown.
         /// </summary>
         /// <returns>JSON Object</returns>
-        public async Task<JObject> Result()
-        {
-            while (await Status() == JobStatus.PENDING)
-            {
-                await Task.Delay(1000);
-            }
-            string result = await FetchResult();
-            var json = JsonConvert.DeserializeObject<JObject>(result);
-            return json;
-        }
+        public async Task<JObject> Result(CancellationToken cancellationToken = default) => await Result<JObject>(cancellationToken);
 
         /// <summary>
         /// Retrieve results. Status must be success or an error will be thrown.
         /// </summary>
         /// <returns>JSON Array</returns>
-        public async Task<JArray> Results()
+        public async Task<JArray> Results(CancellationToken cancellationToken = default) => await Result<JArray>(cancellationToken);
+
+        /// <summary>
+        /// Retrieve results. Status must be success or an error will be thrown.
+        /// </summary>
+        /// <returns>Result of <typeparamref name="TResult"/> type</returns>
+        public async Task<TResult> Result<TResult>(CancellationToken cancellationToken)
         {
             while (await Status() == JobStatus.PENDING)
             {
-               await Task.Delay(1000);
+                await Task.Delay(1000, cancellationToken);
```

Comment:
it's rewritten in V2 of the lib, and there's `checkInterval` parameter for that. I think it can stay here as it is unless explicitly requested by a customer.